src/auth/ainizeAuth.ts

```diff
@@ -0,0 +1,67 @@
+import Ain from '@ainblockchain/ain-js';
+import Ainize from '@ainize-team/ainize-js';
+
+export default class AinizeAuth {
+  private static instance: AinizeAuth;
+  private ain: Ain | null = null;
+  private ainize: Ainize | null = null;
+  private loggedIn: boolean = false;
+
+  private constructor() {}
+
+  static getInstance() {
+    if (!this.instance) {
+      this.instance = new AinizeAuth();
+    }
+    return this.instance;
+  }
+
+  init(ain: Ain, ainize: Ainize) {
+    this.ain = ain;
+    this.ainize = ainize;
+  }
+
+  async login() {
+    if (!this.loggedIn) {
+      const privateKey = this.getPrivateKeyOrThrow();
+      const ainize = this.getAinizeOrThrow();
+      await ainize.login(privateKey);
+      this.loggedIn = true;
+    }
+  }
+
+  async logout() {
+    if (this.loggedIn) {
+      const ainize = this.getAinizeOrThrow();
+      await ainize.logout();
+      this.loggedIn = false;
+    }
+  }
+
+  isLoggedIn() {
+    return this.loggedIn;
+  }
+
+  private getAinOrThrow() {
+    if (!this.ain) {
```

Comment:
(suggestion) how about checking this in getInstance() instead?

Reply:
This code is removed while using object injection.